exec/java-exec/src/main/java/org/apache/drill/exec/planner/logical/partition/FindPartitionConditions.java

```diff
@@ -217,8 +218,21 @@ public Void visitCorrelVariable(RexCorrelVariable correlVariable) {
   }
 
   public Void visitCall(RexCall call) {
-    // assume PUSH until proven otherwise
-    analyzeCall(call, PushDirFilter.PUSH);
+    boolean visited = false;
+    // examine the input of a CAST function; this could be extended for
+    // other functions in the future.
```

Comment:
As the comment indicated, this patch only solves the problem when there is CAST function. Is there any particular reason why other function is not supported? How much effort do we need to get other functions work as well?

I could hit the same problem, with the following filter:

where (dir0 - 1 = 1994 and o_totalprice < 40000) or dir0=1996"